### PR TITLE
test-version-utils: TS5 fixes

### PR DESCRIPTION
**Pull request by DLehenbauer**
Minor typing fixes to unblock TS5 adoption

### CODE REVIEW

File: packages/test/test-version-utils/src/compatConfig.ts [hunk 1]
```diff
@@ -273,7 +273,7 @@ export async function mochaGlobalSetup() {
 		ensurePackageInstalled(baseVersion, value, reinstall),
 	);
 
-	let error: unknown | undefined;
+	let error: unknown;
```
Comment (DLehenbauer) [inline packages/test/test-version-utils/src/compatConfig.ts:276]: Nit: unknown includes undefined

File: packages/test/test-version-utils/src/compatConfig.ts [hunk 2]
```diff
@@ -282,6 +282,7 @@ export async function mochaGlobalSetup() {
 		}
 	}
 	if (error) {
+		// eslint-disable-next-line @typescript-eslint/no-throw-literal -- Rethrows caught value
 		throw error;
```
Comment (DLehenbauer) [inline packages/test/test-version-utils/src/compatConfig.ts:286]: We're re-throwing the caught value.  The type is truly unknown.

File: packages/test/test-version-utils/src/versionUtils.ts [hunk 3]
```diff
@@ -327,7 +326,7 @@ export function getRequestedRange(baseVersion: string, requested?: number | stri
 export function internalSchema(
 	publicVersion: string,
 	internalVersion: string,
-	requested: number | string,
+	requested: number,
```
Comment (DLehenbauer) [inline packages/test/test-version-utils/src/versionUtils.ts:329]: TS5 no longer allows string/number comparisons (e.g., 'requested <= -2').

Reviewing the code, it looks like the caller handles the case where 'requested' is a string:
```ts
export function getRequestedRange(baseVersion: string, requested?: number | string): string {
	if (requested === undefined || requested === 0) {
		return baseVersion;
	}
	if (typeof requested === "string") {
		return requested;
	}
```

**Review by taylorsw04 — APPROVED** [review]

**Pull request merged**